src/org/httpkit/client.clj

```diff
@@ -193,7 +195,9 @@
                                 ;; only the 4 support now
                                 (case as :auto 1 :text 2 :stream 3 :byte-array 4))
         effective-proxy-url (if proxy-host (str proxy-host ":" proxy-port) proxy-url)
-        cfg (RequestConfig. method headers body timeout keepalive effective-proxy-url tunnel?)]
+        conn-timeout (if timeout timeout conn-timeout)
```

Comment:
As I [https://github.com/http-kit/http-kit/pull/315/files#diff-f46a7bd519e56fa14a401cc9d6e2abd2R145](defined default values here), they will always be filled out. So the logic here is that if you provide a general timeout, that will always win. But of course we can change it to be the other way around, leaving default general timeout at 60sec and using the predicate you suggested.